staking/README.md

```diff
@@ -0,0 +1,541 @@
+
+The `PostHandleMsgStakeUpdate` post-handler then derives the new voting power for the validator and persists in the state via the keeper:
+
+```
+// set validator amount
+p, err := helper.GetPowerFromAmount(msg.NewAmount.BigInt())
+if err != nil {
+	return hmCommon.ErrInvalidMsg(k.Codespace(), fmt.Sprintf("Invalid amount %v for validator %v", msg.NewAmount, msg.ID)).Result()
+}
+
+validator.VotingPower = p.Int64()
+
+// save validator
+err = k.AddValidator(ctx, validator)
+if err != nil {
+	k.Logger(ctx).Error("Unable to update signer", "ValidatorID", validator.ID, "error", err)
+	return hmCommon.ErrSignerUpdateError(k.Codespace()).Result()
+}
+```
+
+The `EndBlocker` hook then updates the changes in the validator set.
+
+## How to propose a MsgStakeUpdate transaction
+
+The `bridge` service in an existing validator's heimdall process polls for `StakeUpdate` event periodically and generates and broadcasts the transaction once it detects and parses the event. An existing validator on the network can also leverage the CLI to send the transaction:
+
+```
+heimdallcli tx staking stake-update --proposer <PROPOSER_ADDRESS> --id <VALIDATOR_ID> --tx-hash <ETH_TX_HASH> --staked-amount <STAKED_AMOUNT> --nonce <VALIDATOR_NONCE> --log-index <LOG_INDEX> --block-number <BLOCK_NUMBER>
+```
+
+Or the REST server :
+
+```
+curl -X POST "localhost:1317/staking/stake-update?proposer=<PROPOSER_ADDRESS>&id=<VALIDATOR_ID>&tx-hash=<ETH_TX_HASH>&staked-amount=<STAKED_AMOUNT>&nonce=<VALIDATOR_NONCE>&log-index=<LOG_INDEX>&block-number=<BLOCK_NUMBER>"
+```
+
+## How does a validator update its signer address
+
+A validator can update its signer address in the network by invoking the the `updateSigner` function on the `StakeManager` contract on L1, which emits an `SignerChange` event:
+
+```
+/// @param validatorId unique integer to identify a validator.
+/// @param nonce to synchronize the events in heimdall.
+/// @param oldSigner old address of the validator.
+/// @param newSigner new address of the validator.
+/// @param signerPubkey public key of the validator.
+event SignerChange(
+    uint256 indexed validatorId,
+    uint256 nonce,
+    address indexed oldSigner,
+    address indexed newSigner,
+    bytes signerPubkey
+);
+```
+
+On Heimdall, this event is parsed and a `MsgSignerUpdate` transaction is broadcasted, which is represented by the data structure:
+
+```
+type MsgSignerUpdate struct {
+	From            hmTypes.HeimdallAddress `json:"from"`
+	ID              hmTypes.ValidatorID     `json:"id"`
+	NewSignerPubKey hmTypes.PubKey          `json:"pubKey"`
+	TxHash          hmTypes.HeimdallHash    `json:"tx_hash"`
+	LogIndex        uint64                  `json:"log_index"`
+	BlockNumber     uint64                  `json:"block_number"`
+	Nonce           uint64                  `json:"nonce"`
+}
+```
+where ,
```

Comment:
i'd remove the comma

Reply:
Done: [159a33b](https://github.com/maticnetwork/heimdall/pull/1082/commits/159a33bafa2f43c7cc2ae0f9d57b4257ef036436)